config/dependency.go

```diff
@@ -337,6 +337,7 @@ func getOutputJsonWithCaching(targetConfig string, terragruntOptions *options.Te
 // Clone terragrunt options and update context for dependency block so that the outputs can be read correctly
 func cloneTerragruntOptionsForDependencyOutput(terragruntOptions *options.TerragruntOptions, targetConfig string) (*options.TerragruntOptions, error) {
 	targetOptions := terragruntOptions.Clone(targetConfig)
+	targetOptions.TerraformCommand = "output"
 	targetOptions.TerraformCliArgs = []string{"output", "-json"}
```

Comment:
Should the command be `output` _and_ the args contain `output` too? Or is that redundant?

Reply:
Or should we add a `--terragrunt-skip-hooks` flag and pass that to the dependency module to ensure hooks are disabled?

Reply:
You need both because `TerraformCommand` is used for the hook determination, while `TerraformCliArgs` is what is actually passed to `terraform`. This redundancy is actually what got me confused and caused me to omit one of them...

Reply:
> Or should we add a --terragrunt-skip-hooks flag and pass that to the dependency module to ensure hooks are disabled?

I think it is more intuitive for `output` hooks to run when getting dependencies, given that the mental model is that fetching dependencies is a call to `terragrunt output` in the target module.

Reply:
I could see it either way: i.e., `dependency` = `output`, so run the `output` hooks... Or that hooks should only run when the module is being executed directly, so never run hooks for `dependency`. 

I don't have a strong opinion here, so let's go with what you have.